src/app/loadout/LoadoutDrawerContents.tsx

```diff
@@ -205,3 +212,38 @@ function fillLoadoutFromEquipped(
     }
   }
 }
+
+function fillLoadoutFromInventory(
+  loadout: Loadout,
+  itemsByBucket: { [bucketId: string]: DimItem[] },
+  stores: DimStore[],
+  add: (item: DimItem, e?: MouseEvent, equip?: boolean) => void
+) {
+  if (!loadout) {
+    return;
+  }
+
+  // TODO: need to know which character "launched" the builder
+  const dimStore =
+    (loadout.classType !== DestinyClass.Unknown &&
+      stores.find((s) => s.classType === loadout.classType)) ||
+    getCurrentStore(stores)!;
+  //select items that are in inventory
+  const items = dimStore.items.filter(
+    (item) =>
+      item.bucket.type !== 'Class' && //probably a better way to do this but makes sure it isn't a class
+      itemCanBeInLoadout(item) &&
+      fromEquippedTypes.includes(item.type)
+  );
+
+  for (const item of items) {
+    if (
+      !itemsByBucket[item.bucket.hash] ||
+      !itemsByBucket[item.bucket.hash].some((i) => i.equipped)
+    ) {
```

Comment:
I believe this will add items to the loadout only if there aren't already already any equipped items in the loadout for that slot, which isn't quite what we're going for.